packages/input-option/src/InputOption/InputOption.tsx

```diff
@@ -60,12 +84,18 @@ export const InputOption = Polymorphic<InputOptionProps, 'div'>(
         aria-checked={checked}
         tabIndex={-1}
         className={cx(
+          getContextStyles(renderedContext, state, theme),
+          getTextStyles(renderedContext, state, theme),
           {
-            [inputOptionWedge]: showWedge,
+            [inputOptionWedge]: showWedgeProp,
+            [getWedgeStyles(renderedContext, state, theme)]: shouldRenderWedge,
+            [getHoverStyles(renderedContext, theme)]:
+              !disabled && isInteractive && state !== State.Highlight,
```

Comment:
nit: I would extract his into a separate `shouldShowHoverStyles` var or something like that